src/rules/completed-docs/tagExclusion.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+
+import { Exclusion } from "./exclusion";
+
+export interface ITagExclusionDescriptor {
+    tags: {
+        content: {
+            [i: string]: string;
+        };
+        existence: string[];
+    };
+}
+
+export class TagExclusion extends Exclusion<ITagExclusionDescriptor> {
+    private readonly contentTags = this.descriptor.tags ? this.descriptor.tags.content : {};
+    private readonly existenceTags = new Set(this.descriptor.tags && this.descriptor.tags.existence);
+
+    public excludes(node: ts.Declaration) {
+        const documentationNode = this.getDocumentationNode(node);
+        const tagsWithContents = this.parseTagsWithContents(documentationNode.getFullText());
```

Comment:
Sorry @ajafff, I haven't been able to get that to work. Adding an if check for the node being any of the JSDoc kinds in a new `visitNode` isn't getting triggered. Either I'm missing something obvious or being silly. Are you ok with either taking a look yourself or letting this pass with a followup issue to clean this rule up (switch to the more efficient walker type and use the new APIs instead of type checking)?